packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -20,6 +20,15 @@ Iterable<Element> collectAllElementsFrom(
   Element rootElement, {
   @required bool skipOffstage,
 }) {
+  assert(() {
+    if (rootElement == null) {
+      throw FlutterError(
+        'A root widget has not been attached to the render tree. '
```

Comment:
Looking at how this is used, `rootElement` does not necessary belong to the root widget of the entire render tree (see for example usage in [1]). It's just the root Element of the subtree from which you want to  collect all elements. In other words, I think in theory this could be null even when your render tree has a root widget. Should this assert in this form maybe move to some of the call sites where the actual root element is passed in to this function?

[1] https://github.com/flutter/flutter/blob/87d9c553f4174e9eb97369efa70983c502de0b5d/packages/flutter_test/lib/src/finders.dart#L681-L689